File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  new_array = str.split('')
  reverse_array = []
  new_array.map do { |e| reverse_array.unshift(e) }
  return reverse_array.join
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts "Expecting: 'godym'"
  puts "=>", reverse_string('mydog')

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
# str.split
# map over string_array by unshifting into new aray
# join reverse array into string

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let strArray = str.split('')
  const reverseArray = []
  strArray.map(e => reverseArray.unshift(e))
  let reverseString = reverseArray.join('')
  return reverseString
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
